checkbox-provider-ce-oem/bin/check_vpu_device.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+import os
+import re
+import sys
+import logging
+from pathlib import Path
+
+SOC_ROOT = "/sys/devices/soc0"
+
+
+def init_logger():
+    """
+    Set the logger to log DEBUG and INFO to stdout, and
+    WARNING, ERROR, CRITICAL to stderr.
+    """
+    root_logger = logging.getLogger()
+    root_logger.setLevel(logging.INFO)
+    logger_format = "%(asctime)s %(levelname)-8s %(message)s"
+    date_format = "%Y-%m-%d %H:%M:%S"
+
+    # Log DEBUG and INFO to stdout, others to stderr
+    stdout_handler = logging.StreamHandler(sys.stdout)
+    stdout_handler.setFormatter(logging.Formatter(logger_format, date_format))
+
+    stderr_handler = logging.StreamHandler(sys.stderr)
+    stderr_handler.setFormatter(logging.Formatter(logger_format, date_format))
+
+    stdout_handler.setLevel(logging.DEBUG)
+    stderr_handler.setLevel(logging.WARNING)
+
+    # Add a filter to the stdout handler to limit log records to
+    # INFO level and below
+    stdout_handler.addFilter(lambda record: record.levelno <= logging.INFO)
+
+    root_logger.addHandler(stderr_handler)
+    root_logger.addHandler(stdout_handler)
+
+    return root_logger
+
+
+def get_soc_family():
+    """
+    Read data from /sys/devices/soc0/family
+
+    Raises:
+        SystemExit: If failed to identify SoC family
+
+    Returns:
+        soc_family (str): SoC family.
+    """
+    soc_family = ""
+    path = os.path.join(SOC_ROOT, "family")
```

Comment:
Just wondering, if `pathlib.Path` is used, are we going to replace all the `os.path.join()` and `os.listdir()` with `PurePath.joinpath()` and `Path.iterdir()` ?

Reply:
fixed, thanks.